### PR TITLE
[IMP] mail, *: clean up message component by introducing view model

### PR DESCRIPTION
**Pull request by alexkuhn**
* = hr, hr_holidays, sms, snailmail, website_livechat

Message model incorrectly contained data related to a specific component (only
one) even though there can be multiple message components per message model.

This cascaded to adapting related component/models to the same principle.

Related: https://github.com/odoo/enterprise/pull/21426

Original PR: odoo/odoo#76718
Original Enterprise PR: odoo/enterprise#20964

### CODE REVIEW

**Comment by alexkuhn** [comment]
@seb-odoo cherry-picked to master (was not fw, and no fw in progress)

**Comment by seb-odoo** [comment]
robodoo override=ci/security
The code was already present. The only diff is that `message` was renamed `messageView`

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

File: addons/mail/static/src/components/attachment_image/attachment_image.xml [hunk 1]
```diff
@@ -2,28 +2,29 @@
...
                             <i class="fa fa-spin fa-spinner"/>
                         </div>
                     </t>
                     <div class="o_AttachmentImage_imageOverlay d-flex flex-row justify-content-end position-absolute">
                         <div class="o_AttachmentImage_actions d-flex flex-column">
-                            <t t-if="props.isEditable">
+                            <t t-if="attachmentImage.attachment.isEditable">
                                 <div class="o_AttachmentImage_action o_AttachmentImage_actionUnlink text-center"
-                                     t-att-class="{'o-pretty': attachmentImage.attachment.isLinkedToComposer}" t-on-click="attachmentImage.onClickUnlink" title="Remove"
+                                     t-att-class="{'o-pretty': attachmentImage.attachmentList.composerView}" t-on-click="attachmentImage.onClickUnlink" title="Remove"
```
Comment (alexkuhn) [inline addons/mail/static/src/components/attachment_image/attachment_image.xml:27]: `o-pretty` looks unused

**Pull request closed without merging**